tests/test_environ.py

```diff
@@ -162,17 +165,17 @@ def test_load_static_config(s, exp, loaded, tmpdir, xonsh_builtins):
 
 
 @skip_if_on_unix
-def test_locate_binary_on_windows():
+def test_locate_binary_on_windows(xonsh_builtins):
     files = ('file1.exe', 'FILE2.BAT', 'file3.txt')
     with TemporaryDirectory() as tmpdir:
         for fname in files:
             fpath = os.path.join(tmpdir, fname)
             with open(fpath, 'w') as f:
                 f.write(fpath)
         env = Env({'PATH': [tmpdir], 'PATHEXT': ['.COM', '.EXE', '.BAT']})
```

Comment:
yes `xonsh_builtins` fixture sets `__xonsh_env__` to be an empty dict.
i can replace this part in this test with your snippet.